### PR TITLE
Add user dropdown menu routes

### LINKED ISSUE #32: As a user I want access to my user menu
Rails API endpoint

- [x] We need an api endpoint to give us the user object
    - [x] name
    - [x] avatar URL
    - [x] superuser?
- [ ] We need an api endpoint to give us the user's role

Ember 
- [x] We need a link to the rails route for user settings
- [x] We need a link to the session api to signout
- [x] If superuser, we need a link to rails route for
    - Operations
    - Accounts
    - Rollouts
    - Announcements

### PR DESCRIPTION
**Pull request by zulaica**
- Add isSuperuser attribute to the user model
- Setup a conditional around the superuser menu from showing on non-superusers
- Add links to rails routes for superusers
- Add link to sign out route
- Add site attribute to project factory
- Give nav-user-dropdown access to the project model
- Add rails route to access profile & settings

Fixes #32

### CODE REVIEW

**Comment by zulaica** [comment]
This PR relies on https://github.com/revelation/revelation-project/pull/1100

**Comment by smashedlife** [comment]
This branch isn't mergeable and should be cleaned up

File: app/models/user.js [hunk 1]
```diff
@@ -4,5 +4,6 @@ import Ember from 'ember';
 export default DS.Model.extend({
   name:                 DS.attr('String'),
   avatar:               DS.attr('String'),
-  lastVisitedProjectId: DS.attr('String')
+  lastVisitedProjectId: DS.attr('String'),
+  isSuperuser:          DS.attr('Boolean')
```
Comment (smashedlife) [inline app/models/user.js]: I left this on the API but this shouldn't be here. It should be fed from the API through the role so people can not tamper with it.

**Comment by smashedlife** [comment]
I can't really tell who's code is who's without this being cleaned up. I did see a mix of ```let``` and ```var``` in the same file so you might want to at least clean that up.

**Comment by kscherling** [comment]
Okay this is weird.  Not sure why my project dropdown is in this branch.  It was originally branched form master...

**Comment by kscherling** [comment]
Hmmm, maybe a rebase or merge, it looks like the commit that references my stuff was already merged with master but shows here as well.  @zulaica what were the steps to get this here?

**Comment by zulaica** [comment]
Hmm. Not entirely sure. This branch went through a couple hands. This may have happened after rebasing with Alex's branch which added the user role. I'll be taking a look at this first thing today.

**Pull request merged**